Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/PaintBrushManipulator.cpp

```diff
@@ -509,6 +528,58 @@ namespace AzToolsFramework
         PaintBrushSettingsRequestBus::Broadcast(&PaintBrushSettingsRequestBus::Events::SetColor, brushColor);
     }
 
+    AZStd::vector<float> PaintBrushManipulator::CalculateGaussianWeights(size_t smoothingRadius)
+    {
+        // The Gaussian function is G(x, y) = (1 / 2*pi*sigma^2) * e^(-(x^2 + y^2) / (2*sigma^2)).
+        // This function produces a bell curve symmetric around the origin, where (1 / 2*pi*sigma^2) is the height of the peak,
+        // and sigma (the standard deviation) controls the width of the peak.
+        // x and y are the relative distances from the center.
+
+        // This will contain all of the Gaussian weights that we compute and return.
+        AZStd::vector<float> gaussianWeights;
+
+        // We track the sum of all the weights so that we can normalize them at the end.
+        float sum = 0.0f;
+
+        // Keep a floating-point version of this so that we don't have to do int/float conversions in our loops below.
+        const float radius = aznumeric_cast<float>(smoothingRadius);
+
+        // In practice, only pixels within a distance of 3 * sigma affect the result, so we'll pick a sigma that's 1/3 of the distance
+        // in each direction.
+        const float sigma = AZStd::max(smoothingRadius / 3.0f, 1.0f);
+
+        // We can precompute the peakHeight and expConstant terms outside of the loop.
+        const float peakHeight = (1.0f / (AZ::Constants::TwoPi * sigma * sigma));
```

Comment:
Yeah fair point, I think I prefer it the way I've written it though, since it's nice and easy to see how it matches up with the Gaussian function. These are only calculated once per paint operation, not per-pixel, so I'm not worried about the cost savings from condensing it.